src/screens/Login.js

```diff
@@ -8,25 +8,30 @@ function Login({ navigation }) {
   const [callbacks, setCallbacks] = useState([]);
   const [nxt, setNxt] = useState(null);
 
-  useEffect(() => {
-    async function logout() {
-      await ForgeRockModule.performUserLogout();
-    }
-    logout();
-  }, []);
-
   useEffect(() => {
     (async () => {
-      await ForgeRockModule.frAuthStart();
-      const data = await ForgeRockModule.loginWithoutUI();
-      const next = JSON.parse(data);
-      setNxt(next);
-      setCallbacks(
-        next.callbacks.map((res) => ({
-          ...res,
-          response: JSON.parse(res.response),
-        })),
-      );
+      try {
+        const data = await ForgeRockModule.loginWithoutUI();
+        const next = JSON.parse(data);
+        setNxt(next);
+        setCallbacks(
+          next.callbacks.map((res) => ({
+            ...res,
+            response: JSON.parse(res.response),
+          })),
+        );
+      } catch (err) {
+        await ForgeRockModule.performUserLogout();
+        const data = await ForgeRockModule.loginWithoutUI();
+        const next = JSON.parse(data);
+        setNxt(next);
+        setCallbacks(
+          next.callbacks.map((res) => ({
+            ...res,
+            response: JSON.parse(res.response),
+          })),
+        );
```

Comment:
Can you explain what's happening here? You try the above, but if it fails, you logout and then try again?

Reply:
I can. I was just using this as a way to force myself to logout when I was developing. I changed this to now pass the error message down to the LoginContainer component and render it out.